File: rerun_cpp/src/rerun/archetypes/arrows2d_ext.cpp
#include "arrows2d.hpp"

// #define EDIT_EXTENSION

namespace rerun {
    namespace archetypes {

#ifdef EDIT_EXTENSION
        // <CODEGEN_COPY_TO_HEADER>

        /// Creates new 2D arrows pointing in the given directions, with a base at the origin (0, 0).
        static Arrows2D from_vectors(Collection<components::Vector2D> vectors_) {
            return Arrows2D().with_vectors(vectors_);
        }

        // </CODEGEN_COPY_TO_HEADER>
#endif
    } // namespace archetypes
} // namespace rerun

File: crates/store/re_sdk_types/src/archetypes/arrows2d_ext.rs
use super::Arrows2D;
use crate::components::Vector2D;

impl Arrows2D {
    /// Creates new 2D arrows pointing in the given directions, with a base at the origin (0, 0).
    #[inline]
    pub fn from_vectors(vectors: impl IntoIterator<Item = impl Into<Vector2D>>) -> Self {
        Self::new(vectors)
    }
}
